modules/local/report.nf

```diff
@@ -455,21 +454,19 @@ def recurse_keys(value, keys_rk){
 }
 
 def traverse_values(value, path){
+
     def temp = value
     def value_found = true
-    def ret_val = null
 
     if(path instanceof String){
         temp = temp[path]
         return temp
     }
 
     for(key in path){
-        def key_val
-        if(key.isNumber()){
-            key_val = key.toInteger()
-        }else{
-            key_val = key
+        def key_val = key
+        if(key_val.isNumber()){
+            key_val = key_val.toInteger()
```

Comment:
Is `key_val` here always an integer, or is it possible to be a float?

Reply:
in this case it should always be an integer as it would be getting an index from a JSON file.